src/index.ts

```diff
@@ -1,38 +1,58 @@
+import FormData from 'form-data';
+import { Request, Response } from 'express';
+import fs from "fs";
 import { cwd } from "process";
 import { Environments } from "./enums/environments";
+import db from './utils/db';
+import { get, post, postWithConfig } from './utils/request';
+import { clearBB2Data, getLoggedInUser, IUser, AuthorizationToken } from './utils/db';
+import { generateCodeChallenge, generateRandomState } from './utils/generatePKCE';
 
+const BENE_DENIED_ACCESS = 'access_denied';
 const DEFAULT_CONFIG_FILE_LOCATION = `${cwd()}/.bluebutton-config.json`;
 const SANDBOX_BASE_URL = "https://sandbox.bluebutton.cms.gov";
 const PRODUCTION_BASE_URL = "https://api.bluebutton.cms.gov";
+const GENERAL_DATA_ERR = '{"message": "Unable to load data - query FHIR resource error."}';
 
 type BlueButtonJsonConfig = {
   clientId: string;
   clientSecret: string;
+  callbackUrl: string;
   version?: string;
+  pkce?: boolean;
   environment?: Environments;
 };
 
 type BlueButtonConfig = string | BlueButtonJsonConfig;
 export default class BlueButton {
   clientId: string;
   clientSecret: string;
+  callbackUrl: string;
   version: string;
+  pkce: boolean;
   baseUrl: string;
+  readonly BB2_PATIENT_URL;
```

Comment:
I think these are just constants, no need to create variables for them like this if that's the case. I would just create getter methods like follows:
```
getPatientUrl() {
   return `${this.baseUrl}/${this.version}/fhir/Patient/`
```

Reply:
like the getter/setter